### ISSUE TITLE
How to save a maximum of cookies ?

### ISSUE THREAD

**Issue by maskym** [jjjchens235/bing-rewards#78]
First of all, thank you very much, your tool is a blessing !

Then I use 2FA so I have to manually accept each request. I increased the waiting time if anyone wants to do the same.

The problem with accepting everytime highlights a bigger problem : Microsoft detect bots by listing accounts that connect multiple time with new browser with the same IP.
Saving cookies instead of connecting everytime would mimic more acurately a browser's behavior

I can check it out or I can check out how to make it run in a Docker, as you want !

**Comment by jjjchens235** [comment]
Are you thinking of something like this? https://stackoverflow.com/questions/15058462/how-to-save-and-load-cookies-using-python-selenium-webdriver

That seems reasonable, feel free to create a pull request.

**Comment by maskym** [comment]
Something like this, yes. I already tried something similar but cookies need to have the same domain as the one currently used by selenium so it might be a bit tricky in the end ...

**Comment by AnonymousFalcon** [comment]
You can store all browser-related data easily but passing a `user-data-dir` argument to the driver. This is a better and simpler solution than storing a specific cookie (`KievRPSSecAuth`) or storing all cookies, as it also stores local storage, cache, history, e.t.c.

This also fixes https://github.com/jjjchens235/bing-rewards/issues/50 since there's no need to store passwords anymore, just the cookie

```python
from selenium import webdriver
from selenium.webdriver.chrome.options import Options

options = Options()
options.add_argument("user-data-dir=stored_browser_data")

driver = webdriver.Chrome(options=options)
```

**Comment by jjjchens235** [comment]
@AnonymousFalcon 
This is so cool, thanks for sharing!

I'll have to figure out a way to integrate this into the current program. 

Currently, the program allows the user to input a different account/pw on the command line than from the one that is stored.

Ideally, this cookie option would be the default, but if they enter the --email argument, then it would use a cookie-less browser.

**Comment by jjjchens235** [comment]
@AnonymousFalcon 
I have implemented. 
Note that the user still has to enter their credentials in the setup process. My thought process was that after X days, even with cookies, Bing will ask the user to enter their credentials and I didn't want the program to fail on this condition.
Thanks again for suggesting this useful feature.

**Comment by AnonymousFalcon** [comment]
@jjjchens235

Even after a certain number of days, bing shouldn't ask the user to enter their credentials.
If there was a previous implementation with just the authentication cookie, this would've been a problem if you saved the token, since the access tokens expire.
However, with `user-data-dir`, everything is stored, including the refresh token. **Bing should automatically refresh the access token when its close to expiring**.

Nevertheless, its not a bad idea to store the credentials anyway in case the token renewal somehow fails. However it would be better if this was optional so as to fix #50

**Comment by jjjchens235** [comment]
All fair points, but I decided I'm not going to add anything extra. Not opposed to someone submitting a PR for this though. Thanks again for your help.
